### DISCUSSION TITLE
Log culling/rotating?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#884] [Q&A]
```
-rw-r--r--  1 meta  staff   116K Oct 29 07:38 data.db
-rw-r--r--  1 meta  staff   796K Oct 29 07:38 logs.db
-rw-r--r--  1 meta  staff    32K Oct 30 03:18 logs.db-shm
-rw-r--r--  1 meta  staff   1.8M Oct 30 03:18 logs.db-wal
```

What is the best way to trim the log files? Or is there a feature planned for this?

**Comment by ganigeorgiev** [comment]
They should be automatically rotated on the first request that comes after the logs retention days limit (could be changed from "Admin UI > Settings > Application > Logs max days retention"; if it is `0` there will be no request logs).

You can also delete old log requests manually via `app.LogsDao().DeleteOldRequests(createdBefore time.Time)`, eg.:

```go
// keep only the logs from the last 3 days
app.LogsDao().DeleteOldRequests(time.Now().AddDate(0, 0, -3))
```

**Reply by benallfree** [reply]
Ok thanks.

Maybe this is a separate discussion, but there are a few settings I'd like to lock down/hide/make read-only in pockethost.io so they can't be modified via the admin. I'm curious how I might approach that.

Examples:

* log limits
* SMTP credentials
* S3 credentials

I suppose I could run custom pocketbase builds that override all these, but I was wondering what you thought.

**Reply by ganigeorgiev** [reply]
The Admin UI currently can't be customized dynamically, but you can restrict/override settings changes by attaching to the `OnSettingsBeforeUpdateRequest` hook (this is how I prevent some settings changes for the live demo):

```go
app.OnSettingsBeforeUpdateRequest().Add(func(e *core.SettingsUpdateEvent) error {
    log.Println(e.OldSettings)
    log.Println(e.NewSettings)

    if e.NewSettings.Smtp.Enabled {
        return apis.NewBadRequestError("Cannot enable SMTP for the demo.", nil)
    }

    return nil
});
```
